### ISSUE TITLE
insane 5 am rambling time

### ISSUE THREAD

**Issue by onkoe** [onkoe/nqcsv#1]
ok i thought of a potentially cool way to do some validation

basically, make the serialization methods change self to have a different 'owned' phantomdata

make the user call a `serializer.finish_record()` method to stop the record. howeber, this can only be called when it owns a `FilledRecord` with its PhantomData, performing compile-time checks on the _number_ of successfully-filled fields in each record.

in other words, deny the user from making mistakes (and avoid those awful runtime errors in the otherwise chill `csv` crate)
